src/lwc/gePaymentGatewayManagement/gePaymentGatewayManagement.js

```diff
@@ -0,0 +1,177 @@
+ *       notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in the
+ *       documentation and/or other materials provided with the distribution.
+ *     * Neither the name of Salesforce.org nor the names of
+ *       its contributors may be used to endorse or promote products derived
+ *       from this software without specific prior written permission.
+ *
+ *     THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *     "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *     LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *     FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *     COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *     INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *     BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *     LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *     CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *     LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *     ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *     POSSIBILITY OF SUCH DAMAGE.
+ */
+import { LightningElement } from 'lwc';
+import { buildErrorMessage } from 'c/utilTemplateBuilder';
+import { isEmpty } from 'c/utilCommon';
+
+import messageLoading from '@salesforce/label/c.labelMessageLoading';
+import insufficientPermissions from '@salesforce/label/c.commonInsufficientPermissions';
+import commonAdminPermissionErrorMessage from '@salesforce/label/c.commonAdminPermissionErrorMessage';
+
+import setGatewayId from '@salesforce/apex/PS_GatewayManagement.setGatewayId';
+import getGatewayIdFromConfig from '@salesforce/apex/PS_GatewayManagement.getGatewayIdFromConfig';
+import checkForElevateCustomer from '@salesforce/apex/PS_GatewayManagement.isElevateCustomer';
+import checkForSystemAdmin from '@salesforce/apex/PS_GatewayManagement.isSystemAdmin';
+
+export default class GePaymentGatewayManagement extends LightningElement {
+
+    showSpinner;
+    gatewayId;
+
+    isReadOnly = true;
+    errorMessage;
+
+    isElevateCustomer;
+    isSystemAdmin;
+    hasAccess;
+
+    CUSTOM_LABELS = { messageLoading, insufficientPermissions, commonAdminPermissionErrorMessage };
+
+    async connectedCallback() {
+        try {
+            this.isSystemAdmin = await checkForSystemAdmin();
+            this.isElevateCustomer = await checkForElevateCustomer();
+
+            this.hasAccess = !!(this.isElevateCustomer && this.isSystemAdmin);
+        } catch(ex) {
+            this.errorMessage = buildErrorMessage(ex);
+            this.isError = true;
+        }
+
+        if (this.hasAccess) {
+            this.getGatewayId();
```

Comment:
I think we're missing the `async` keyword here and also not handling any error scenarios with this call. Do you also think we could call `await getGatewayIdFromConfig()` directly here instead since we're not using `getGatewayIdFromConfig` anywhere else. Also I would love to see us combine all three server calls in the `connectedCallback` into one instead of making three server trips. What are your thoughts?

Reply:
@voduyemi Regarding the error scenarios, since there aren't any specific error scenarios we would anticipate here, @kenlewis-sfdo mentioned we should probably just remove the try/catch which I agree with since we want to avoid any more generic error handling. There's really no way we could get a failure here just retrieving the gatewayId. Also, since the `async` keyword is on the connected callback function, this function isn't one we have to set an `await` on since there's no other logic after it. It will just populate the field on the component load when the call finishes